s4e-web/src/app/utils/miscellaneous/miscellaneous.ts

```diff
@@ -123,3 +124,17 @@ export function toHashMap<T>(list: T[], key: keyof T): HashMap<T> {
     return prev;
   }, {});
 }
+
+export function isEmptyObject(obj: object): boolean {
+  return Object.keys(obj).length === 0 && obj.constructor === Object;
+}
+
+export function stripNulls(obj: object): object {
+  const r = {};
+
+  Object.entries(obj).filter(([k, v]) => v != null).forEach(([k, v]) => {
+    r[k] = v
+  });
+
+  return r;
```

Comment:
It will be safer to `deep clone` object and then remove unnecessary keys from it. In case if values won't be simple types they will be passed as a reference.

const objClone = JSON.parse(JSON.stringify(obj));
Object.keys(objClone).foreach((key) => objClone[key] === null ? delete(objClone[key]) : null);
return objClone

Reply:
This is acceptable, doing deep copy this way has a lot of it's own problems (all operators working on Array (e.g. `filter` works the same it does not deep copy objects). The idea is to just strip nulls, not create copy of the object.